test/scales/linearScaleTests.ts

```diff
@@ -1,134 +1,267 @@
+        };
+
+        scale.domain([1, 2]);
+        assert.deepEqual(scale.domain(), [1, 2], "initial domain is set");
+
+        warningCalled = false;
+        scale.domain([5, Infinity]);
+        assert.deepEqual(scale.domain(), [1, 2], "Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to Infinity");
+
+        warningCalled = false;
+        scale.domain([-Infinity, 5]);
+        assert.deepEqual(scale.domain(), [1, 2], "-Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to -Infinity");
+
+        warningCalled = false;
+        scale.domain([NaN, 7]);
+        assert.deepEqual(scale.domain(), [1, 2], "NaN containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to NaN");
+
+        warningCalled = false;
+        scale.domain([-1, 5]);
+        assert.deepEqual(scale.domain(), [-1, 5], "Regular domains still accepted");
+        assert.isFalse(warningCalled, "a warning is not thrown when correct domains are given");
+
+        Plottable.Utils.Window.warn = oldWarn;
+      });
+
+      it("accepts NaN or Infinity as range", () => {
+        scale.range([5, Infinity]);
+        assert.deepEqual(scale.range(), [5, Infinity], "Infinity accepted as part of range");
+        scale.range([-Infinity, 5]);
+        assert.deepEqual(scale.range(), [-Infinity, 5], "-Infinity accepted as part of range");
+        scale.range([NaN, 7]);
+        assert.deepEqual(scale.range(), [NaN, 7], "NaN accepted as part of range");
+      });
+
+      it("can receive custom tick generator", () => {
+        scale.domain([0, 10]);
+        let defaultTicks = scale.ticks();
+        assert.strictEqual(defaultTicks.length, 11, "ticks were generated correctly with default generator");
+        scale.tickGenerator((scale) => scale.defaultTicks().filter(tick => tick % 3 === 0));
+        let customTicks = scale.ticks();
+        assert.deepEqual(customTicks, [0, 3, 6, 9], "ticks were generated correctly with custom generator");
+      });
     });
 
-    it("domain can't include NaN or Infinity", () => {
-      let scale = new Plottable.Scales.Linear();
-      scale.domain([0, 1]);
-      scale.domain([5, Infinity]);
-      assert.deepEqual(scale.domain(), [0, 1], "Infinity containing domain was ignored");
-      scale.domain([5, -Infinity]);
-      assert.deepEqual(scale.domain(), [0, 1], "-Infinity containing domain was ignored");
-      scale.domain([NaN, 7]);
-      assert.deepEqual(scale.domain(), [0, 1], "NaN containing domain was ignored");
-      scale.domain([-1, 5]);
-      assert.deepEqual(scale.domain(), [-1, 5], "Regular domains still accepted");
+    describe("Auto Domaining", () => {
+      let scale: Plottable.Scales.Linear;
+
+      beforeEach(() => {
+        scale = new Plottable.Scales.Linear();
+      });
+
+      it("auto domains by default", () => {
```

Comment:
Technically this tests the default domain with no extents, rather than that it auto domains by default...